### ISSUE TITLE
:create suffix only creates dirs

### ISSUE THREAD

**Issue by bam80** [flatpak/flatpak#2302]
> For --filesystem=FS option, ":create suffix indicates that the location will be read-write and created if it doesn't exist".

If it doesn't exist, it will always be treated as directory, and created as such. There is no way to point that last part of the created location should be a file.

**Comment by bam80** [comment]
@matthiasclasen I think yes, because otherwise you can't bind mount a file if it doesn't exist

**Comment by matthiasclasen** [comment]
I would suggest to patch the app. It should not look at either ~/.vuescanrc or ~/.vuescan/vuescanrc
but at $XDG_CONFIG_HOME/vuescanrc

**Comment by TingPing** [comment]
Most of the bad behaving apps are proprietary.

**Comment by TingPing** [comment]
Regarding vuescan this is probably fine: https://github.com/flathub/com.hamrick.VueScan/pull/1

**Comment by ritzk** [comment]
@TingPing  This does not help, if the file ( ~/.vuescanrc) does not exist.  

Either
- flatpak will need to create an empty file.
- or, Alternatively, one can pass "--env=HOME=~/.vuescan/"  to workaround this issue.

@matthiasclasen  proprietary application. The author has no plans to update this behaviour.

**Comment by FakeShemp** [comment]
Yes, I had to revert https://github.com/flathub/com.hamrick.VueScan/pull/1 since it created an empty directory named `.vuescanrc` when actually, an empty file is needed instead.

The current manifest doesn't actually work as intended either, because the user needs to create the empty file manually at `~/.vuescanrc` since `:create` and such always assumes folders are to be created.

**Comment by FakeShemp** [comment]
If `persist` could create a file, that would be preferable.

**Comment by TingPing** [comment]
> If persist could create a file, that would be preferable.

You are going backwards, `--persist` is to avoid escaping the sandbox and `--filesystem` is to escape the sandbox. The former should never need to create anything because the only reason `:create` exists is the technical detail of how bind mounts work pulling in data from outside the sandbox.

**Comment by matthiasclasen** [comment]
well, --persist bind mounts too, no ?

**Comment by TingPing** [comment]
Hmm, I suppose it does, but clearly the filename doesn't have to exist ahead of time or else it would be useless. Would have to look into how that works.
